File: 9-palindrome-number/9-palindrome-number.c
bool isPalindrome(int x){
    int tmp1 = x;
    long tmp2 = 0;
    
    if (x < 0)
        return false;
    else
    {
        while (tmp1 > 0)
        {
            tmp2 = tmp2 * 10 + (tmp1 % 10);
            tmp1 /= 10;
        }
        if (tmp2 == x)
            return true;
        else
            return false;
    }
}

File: 9-palindrome-number/9-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0) return false;
        
        int digit = digitsNumber(x);
        int tmp = x; 
        long s = 0;
        
        for(; digit >= 1; digit--){
            s += (tmp % 10) * Math.pow(10, digit-1);
            tmp /= 10;
        }
        return (x == s);
    }
    
    public int digitsNumber(int k){
        int digit = 1;
        
        while(k > 9){
            k /= 10;
            ++digit;
        }
        
        return digit;
    }
}